File: Platforms/Windows/Reign.Video.D3D9/Mesh.cpp
#include "pch.h"
#include "Mesh.h"

namespace Reign
{namespace Video
{namespace D3D9
{
	#pragma region Constructors
	Mesh::Mesh(BinaryReader^ reader, ModelI^ model, int classicInstanceCount)
	: MeshI(reader, model, classicInstanceCount)
	{
		
	}

	BufferLayoutDescI^ Mesh::createBufferLayoutDesc(List<BufferLayoutElement>^ elements)
	{
		return gcnew BufferLayoutDesc(elements);
	}

	VertexBufferI^ Mesh::createVertexBuffer(DisposableI^ parent, BufferLayoutDescI^ layoutDesc, BufferUsages usage, VertexBufferTopologys topology, array<float>^ vertices)
	{
		return gcnew Reign::Video::D3D9::VertexBuffer(parent, layoutDesc, usage, topology, vertices);
	}

	IndexBufferI^ Mesh::createIndexBuffer(DisposableI^ parent, BufferUsages usage, array<int>^ indices, bool _32BitIndices)
	{
		return gcnew Reign::Video::D3D9::IndexBuffer(parent, usage, indices, _32BitIndices);
	}
	#pragma endregion
}
}
}

File: Platforms/Common/Reign.Video.XNA/Model/Mesh.cs
using System;
using Reign.Core;
using System.Collections.Generic;
using System.IO;

namespace Reign.Video.XNA
{
	public class Mesh : MeshI
	{
		#region Constructors
		public Mesh(BinaryReader reader, ModelI model, int classicInstanceCount)
		: base(reader, model, classicInstanceCount)
		{
			
		}

		protected override BufferLayoutDescI createBufferLayoutDesc(List<BufferLayoutElement> elements)
		{
			return new BufferLayoutDesc(elements);
		}

		protected override VertexBufferI createVertexBuffer(DisposableI parent, BufferLayoutDescI layoutDesc, BufferUsages usage, VertexBufferTopologys topology, float[] vertices)
		{
			return new VertexBuffer(parent, layoutDesc, usage, topology, vertices);
		}

		protected override IndexBufferI createIndexBuffer(DisposableI parent, BufferUsages usage, int[] indices, bool _32BitIndices)
		{
			return new IndexBuffer(parent, usage, indices, _32BitIndices);
		}
		#endregion
	}
}